### ISSUE TITLE
DefaultYouTrack should accept instances of Login in ctor, not Session

### ISSUE THREAD

**Issue by llorllale** [llorllale/youtrack-api#161]
Refactor `DefaultYouTrack` so that the following

```java
final YouTrack youtrack = new DefaultYouTrack(
    new PermanentToken("token").login()
);
```

would instead be written as:

```java
final YouTrack youtrack = new DefaultYouTrack(
    new PermanentToken("token")
);
```

`AuthenticationException` is an `IOException` so it would not break any API contracts.

Motivation: elimination of procedural code.

**Comment by 0pdd** [comment]
@llorllale the puzzle [#166](https://github.com/llorllale/youtrack-api/issues/166) is still not solved.

**Comment by 0pdd** [comment]
@llorllale the puzzle [#196](https://github.com/llorllale/youtrack-api/issues/196) is still not solved; solved: [#166](https://github.com/llorllale/youtrack-api/issues/166).

**Comment by 0pdd** [comment]
@llorllale all 2 puzzles are solved here: [#166](https://github.com/llorllale/youtrack-api/issues/166), [#196](https://github.com/llorllale/youtrack-api/issues/196).
